🔥[🐛] Crash - Using in-app-messaging (or remote-config) without analytics

<!---
**Note <= v5 is deprecated, v5 issues are unlikely to get attention. Feel free to ask in discussions instead.**

Hello there you awesome person;

Please note that the issue list of this repo is exclusively for bug reports;

1) For feature requests, questions and general support please use [GitHub Discussions](https://github.com/invertase/react-native-firebase/discussions).
2) If this is a setup issue then please make sure you've correctly followed the setup guides, most setup issues such as 'duplicate dex files', 'default app has not been initialized' etc are all down to an incorrect setup as the guides haven't been correctly followed.
-->

<!-- NOTE: You can change any of the `[ ]` to `[x]` to mark an option(s) as selected -->

<!-- PLEASE DO NOT REMOVE ANY SECTIONS FROM THIS ISSUE TEMPLATE   -->
<!--   Leave them as they are even if they're irrelevant to your issue -->

## Issue

<!-- Please describe your issue here --^ and provide as much detail as you can. -->
<!-- Include code snippets that show your usages of the library in the context of your project. -->
<!-- Snippets that also show how and where the library is imported in JS are useful to debug issues relating to importing or methods not found issues -->

Hello guys. We updated to firebase 10.5.0 (for app package and 10.5.1 for the other packages) and rollout in production. For the first time our play console registered this crash on opening the app. It impacted ~ 1k/20k users. It was something that we couldn't replicate with our devices & emulators.

This is the stack trace from play console

```
java.lang.RuntimeException: 
  at android.app.ActivityThread.installProvider (ActivityThread.java:6747)
  at android.app.ActivityThread.installContentProviders (ActivityThread.java:6289)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6204)
  at android.app.ActivityThread.access$1200 (ActivityThread.java:237)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1785)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7050)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:494)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:965)
Caused by: com.google.firebase.components.MissingDependencyException: 
  at com.google.firebase.components.ComponentRuntime.b (ComponentRuntime.java:125)
  at com.google.firebase.components.ComponentRuntime.a (ComponentRuntime.java:130)
  at com.google.firebase.components.ComponentRuntime.<init> (ComponentRuntime.java:106)
  at com.google.firebase.components.ComponentRuntime.<init> (ComponentRuntime.java)
  at com.google.firebase.components.ComponentRuntime.<init> (ComponentRuntime.java)
  at com.google.firebase.FirebaseApp.<init> (FirebaseApp.java:140)
  at com.google.firebase.FirebaseApp.a (FirebaseApp.java:60)
  at com.google.firebase.FirebaseApp.a (FirebaseApp.java:37)
  at com.google.firebase.provider.FirebaseInitProvider.onCreate (FirebaseInitProvider.java:4)
  at android.content.ContentProvider.attachInfo (ContentProvider.java:1940)
  at android.content.ContentProvider.attachInfo (ContentProvider.java:1915)
  at com.google.firebase.provider.FirebaseInitProvider.attachInfo (FirebaseInitProvider.java:15)
  at android.app.ActivityThread.installProvider (ActivityThread.java:6742)
  at android.app.ActivityThread.installContentProviders (ActivityThread.java:6289)
  at android.app.ActivityThread.handleBindApplication (ActivityThread.java:6204)
  at android.app.ActivityThread.access$1200 (ActivityThread.java:237)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1785)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7050)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:494)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:965)
```

### What we updated:
1. firebase:
<img width="1443" alt="Screenshot 2021-01-25 at 23 23 12" src="https://user-images.githubusercontent.com/35928720/105818249-5ab2ed80-5fbf-11eb-9834-157f64d60d84.png">

2. versions in android/build.gradle (may be related):
<img width="1434" alt="Screenshot 2021-01-26 at 10 15 54" src="https://user-images.githubusercontent.com/35928720/105818511-b1b8c280-5fbf-11eb-8211-d3098f20ccd5.png">


Then we got a more specific stacktrace from a beta tester:
![Screenshot 2021-01-26 at 10 13 04](https://user-images.githubusercontent.com/35928720/105818044-16275200-5fbf-11eb-92ad-822eca9f400e.png)

Reached out to the tester and downgrade firebase versions from `package.json` to previous values. It works now for him.

But I would not jump into conclusions as I researched this a lot. I'm not sure if it's a firebase problem(probably native implementation) or a cache problem with our project

Other issues i find related:
1. https://github.com/invertase/react-native-firebase/issues/3670 - Almost same exception but is not satisfied with Crashlytics, instead of InAppMessaging.

---

## Project Files

<!-- Provide the contents of key project files which will help to debug -->
<!--     For Example: -->
<!--        - iOS: `Podfile` contents. -->
<!--        - Android: `android/build.gradle` contents. -->
<!--        - Android: `android/app/build.gradle` contents. -->
<!--        - Android: `AndroidManifest.xml` contents. -->

<!-- ADD THE CONTENTS OF THE FILES IN THE PROVIDED CODE BLOCKS BELOW -->


### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
{
"dependencies": {
    "@invertase/react-native-apple-authentication": "1.1.2",
    "@react-native-community/async-storage": "1.12.1",
    "@react-native-community/clipboard": "1.5.1",
    "@react-native-community/google-signin": "5.0.0",
    "@react-native-community/netinfo": "5.9.7",
    "@react-native-firebase/app": "10.5.0",
    "@react-native-firebase/crashlytics": "10.5.1",
    "@react-native-firebase/dynamic-links": "10.5.1",
    "@react-native-firebase/in-app-messaging": "10.5.1",
    "@react-native-firebase/messaging": "10.5.1",
    "@react-native-firebase/remote-config": "10.5.1",
    "@sentry/react-native": "2.1.0",
    "@types/detox": "16.4.3",
    "axios": "0.21.0",
    "date-fns": "2.16.1",
    "fast-memoize": "2.5.2",
    "jetifier": "1.6.6",
    "lottie-react-native": "3.5.0",
    "patch-package": "6.2.2",
    "postinstall-postinstall": "2.1.0",
    "prop-types": "15.7.2",
    "query-string": "6.13.7",
    "react": "16.13.1",
    "react-native": "0.63.2",
    "react-native-amplitude-analytics": "0.2.11",
    "react-native-animatable": "1.3.3",
    "react-native-code-push": "6.4.0",
    "react-native-contacts": "6.0.3",
    "react-native-draggable-flatlist": "2.5.0",
    "react-native-exception-handler": "2.10.9",
    "react-native-fast-image": "8.3.3",
    "react-native-fbsdk": "3.0.0",
    "react-native-gesture-handler": "1.8.0",
    "react-native-image-crop-picker": "0.34.1",
    "react-native-image-viewing": "0.2.0",
    "react-native-indicators": "0.17.0",
    "react-native-intercom-native": "1.0.1",
    "react-native-iphone-x-helper": "1.3.1",
    "react-native-keyboard-aware-scroll-view": "0.9.3",
    "react-native-keyboard-spacer": "0.4.1",
    "react-native-keychain": "3.1.3",
    "react-native-linear-gradient": "2.5.6",
    "react-native-modalize": "2.0.6",
    "react-native-navigation": "6.11.0",
    "react-native-navigation-hooks": "6.0.2",
    "react-native-rate": "1.2.4",
    "react-native-reanimated": "1.9.0",
    "react-native-restart": "0.0.17",
    "react-native-share": "5.1.0",
    "react-native-sms": "1.11.0",
    "react-native-sound": "0.11.0",
    "react-native-svg": "12.1.0",
    "react-native-swiper": "1.6.0",
    "react-native-v8": "0.63.2-patch.1",
    "react-native-version-check": "3.4.2",
    "react-native-walkthrough-tooltip": "1.1.11",
    "react-native-webview": "10.10.2",
    "react-redux": "7.2.2",
    "redux": "4.0.5",
    "redux-persist": "6.0.0",
    "redux-persist-filesystem-storage": "3.0.0",
    "redux-persist-transform-filter": "0.0.20",
    "redux-thunk": "2.3.0",
    "reselect": "4.0.0",
    "rn-fetch-blob": "0.12.0",
    "rn-placeholder": "3.0.3",
    "singular-react-native": "^1.1.6",
    "ts-jest": "26.4.4",
    "v8-android-nointl": "8.84.0"
  },
}
```

#### `firebase.json` for react-native-firebase v6:

```json
{
  "react-native": {
    "messaging_android_notification_channel_id": "high-priority"
  }
}

```

</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

<!--- Mark any options that apply below -->

- [x] my application is an AndroidX application?
- [ ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [x] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    ext {
        RNNKotlinVersion = '1.3.61'
        buildToolsVersion = '29.0.2'
        minSdkVersion = 21
        compileSdkVersion = 29
        targetSdkVersion = 29
        supportLibVersion = "28.0.0"
        googlePlayServicesVersion = "16.1.0"
        kotlinVersion = '1.3.41'
        RNNKotlinStdlib = "kotlin-stdlib-jdk8"

    }
    ext.detoxKotlinVersion = ext.kotlinVersion

    repositories {
        mavenLocal()
        mavenCentral()
        google()
        jcenter()
        maven {
            url 'https://maven.fabric.io/public'
        }
        // From react-native-image-picker-crop
        maven { url 'https://maven.google.com' }

        // From react-native-image-picker-crop
        maven { url "https://www.jitpack.io" }

        
    }
    dependencies {
        classpath('com.android.tools.build:gradle:4.0.2')
        classpath 'com.google.gms:google-services:4.3.3'
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
        classpath 'de.mannodermaus.gradle.plugins:android-junit5:1.4.2.1'
        classpath 'de.undercouch:gradle-download-task:3.4.3'
        classpath 'com.google.firebase:firebase-crashlytics-gradle:2.2.0'


        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            //url("$rootDir/../node_modules/react-native/android")

            // Replace AAR from original RN with AAR from react-native-v8
            url("$rootDir/../node_modules/react-native-v8/dist")
        }
        maven {
            // Android JSC is installed from npm
            // url("$rootDir/../node_modules/jsc-android/dist")

            // prebuilt libv8android.so
            url("$rootDir/../node_modules/v8-android-nointl/dist")
        }
        maven {
            url "$rootDir/../node_modules/detox/Detox-android"
        }
        google()
        mavenCentral()
        jcenter()
        maven { url 'https://www.jitpack.io' }

        maven { url 'http://maven.singular.net/' }

        // force dependency versions on all subprojects
        configurations.all {
            resolutionStrategy {
                // force soloader 0.9.0 for x86 arhitecture crash
                force "com.facebook.soloader:soloader:0.9.0"
			}
		}
    }
}

subprojects { subproject ->
    afterEvaluate {
        if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
            android {
                variantFilter { variant ->
                    def names = variant.flavors*.name
                    if (names.contains("reactNative51") || names.contains("reactNative55") || names.contains("reactNative56") || names.contains("reactNative57") || names.contains("reactNative57_5") || names.contains("reactNative57WixFork") || names.contains("ReactNative57Wix") || names.contains("reactNative60")) {
                        setIgnore(true)
                    }
                }
            }
        }
    }
}
```

#### `android/app/build.gradle`:

```groovy
apply plugin: "com.android.application"

import com.android.build.OutputFile

project.ext.react = [
    enableHermes : false,  // clean and rebuild if changing
]

apply from: "../../node_modules/react-native/react.gradle"
apply from: "../../node_modules/react-native-code-push/android/codepush.gradle"
apply from: "../../node_modules/@sentry/react-native/sentry.gradle"

def enableSeparateBuildPerCPUArchitecture = false

def enableProguardInReleaseBuilds = false


def jscFlavor = 'org.webkit:android-jsc:+'

def enableHermes = project.ext.react.get("enableHermes", false)

android {
    signingConfigs {
        release {
            if (project.hasProperty('MYAPP_UPLOAD_STORE_FILE')) {
                storeFile file(MYAPP_UPLOAD_STORE_FILE)
                storePassword MYAPP_UPLOAD_STORE_PASSWORD
                keyAlias MYAPP_UPLOAD_KEY_ALIAS
                keyPassword MYAPP_UPLOAD_KEY_PASSWORD
            }
        }
    }
    compileSdkVersion rootProject.ext.compileSdkVersion
    buildToolsVersion rootProject.ext.buildToolsVersion

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    defaultConfig {
        applicationId "com.xxxxxxxxx"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode 3146135
        versionName "5.1.0"
        multiDexEnabled true
        missingDimensionStrategy "RNN.reactNativeVersion", "reactNative62"
        renderscriptTargetApi rootProject.ext.targetSdkVersion
        renderscriptSupportModeEnabled true
        resConfigs "en", "US"
        testBuildType System.getProperty('testBuildType', 'debug')
        testInstrumentationRunner 'androidx.test.runner.AndroidJUnitRunner'
        vectorDrawables.useSupportLibrary = true
    }

    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
        }
    }
    buildTypes {
        debug {
        }
        release {
            signingConfig signingConfigs.release
            // optimization
            minifyEnabled true
            shrinkResources true
            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
        }
    }

    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            def versionCodes = ["armeabi-v7a": 1, "x86": 2, "arm64-v8a": 3, "x86_64": 4]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }

    packagingOptions {       
        exclude "**/libjsc.so"
    }
}

dependencies {
    def multidex_version = "2.0.1"
    def appcompat_version = "1.0.2"

    implementation fileTree(dir: "libs", include: ["*.jar"])

    if (enableHermes) {
        def hermesPath = "../../node_modules/hermes-engine/android/"
        debugImplementation files(hermesPath + "hermes-debug.aar")
        releaseImplementation files(hermesPath + "hermes-release.aar")
    } else {
        implementation jscFlavor
    }

    implementation project(':react-native-exception-handler')
    implementation project(':react-native-version-check')  

    implementation "com.google.android.gms:play-services-base:${rootProject.ext.googlePlayServicesVersion}"

    implementation 'com.amplitude:android-sdk:2.19.1' // native sdk of amplitude
    implementation project(':react-native-amplitude-analytics')
    implementation 'com.facebook.android:facebook-android-sdk:[4,5)'
    implementation project(':react-native-keychain')
    implementation project(':react-native-linear-gradient')
    implementation "androidx.appcompat:appcompat:$appcompat_version"
    implementation "com.facebook.react:react-native:+"  // From node_modules

    implementation 'org.chromium:v8-android:+'

    implementation "androidx.swiperefreshlayout:swiperefreshlayout:1.0.0"
    debugImplementation("com.facebook.flipper:flipper:${FLIPPER_VERSION}") {
      exclude group:'com.facebook.fbjni'
    }
    debugImplementation("com.facebook.flipper:flipper-network-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
    }
    debugImplementation("com.facebook.flipper:flipper-fresco-plugin:${FLIPPER_VERSION}") {
        exclude group:'com.facebook.flipper'
    }

    // Force webkit version. Needed for editor
    implementation 'androidx.webkit:webkit:1.2.0'

    implementation project(':react-native-navigation')
    implementation 'com.android.support:design:27.1.0'
    implementation "androidx.multidex:multidex:$multidex_version"
    implementation('com.crashlytics.sdk.android:crashlytics:2.9.9@aar') {
        transitive = true
    }

    implementation 'com.facebook.soloader:soloader:0.9.0+'

    androidTestImplementation('com.wix:detox:+') { transitive = true }
    androidTestImplementation 'junit:junit:4.12'
}

task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}

apply plugin: 'com.google.gms.google-services'
apply plugin: 'com.google.firebase.crashlytics'

apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
configurations {
    all*.exclude group: 'com.google.guava', module: 'listenablefuture'
}
```

#### `android/settings.gradle`:

```groovy
rootProject.name = 'xxxxxxxxx'
include ':react-native-webview'
project(':react-native-webview').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-webview/android')
apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)

include ':app', ':react-native-code-push'
project(':react-native-code-push').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-code-push/android/app')
include ':react-native-exception-handler'
project(':react-native-exception-handler').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-exception-handler/android')
include ':react-native-version-check'
project(':react-native-version-check').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-version-check/android')
include ':react-native-amplitude-analytics'
project(':react-native-amplitude-analytics').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-amplitude-analytics/android')
include ':react-native-keychain'	
project(':react-native-keychain').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-keychain/android')
include ':react-native-linear-gradient'
project(':react-native-linear-gradient').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-linear-gradient/android')
include ':singular-react-native'
project(':singular-react-native').projectDir = new File(rootProject.projectDir, '../node_modules/singular-react-native/android')
include ':react-native-navigation'
project(':react-native-navigation').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-navigation/lib/android/app/')

include ':app'
```

#### `MainApplication.java`:

```java
package com.xxxxxxxx;

import android.content.Context;

import androidx.multidex.MultiDex;

import com.microsoft.codepush.react.CodePush;

import com.BV.LinearGradient.LinearGradientPackage;
import com.facebook.CallbackManager;
import com.facebook.appevents.AppEventsLogger;
import com.facebook.react.PackageList;
import com.facebook.react.ReactInstanceManager;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.reactnativenavigation.NavigationApplication;
import com.reactnativenavigation.react.NavigationReactNativeHost;
import com.reactnativenavigation.react.ReactGateway;

import java.lang.reflect.InvocationTargetException;
import java.util.List;

import com.facebook.react.modules.i18nmanager.I18nUtil;

public class MainApplication extends NavigationApplication {

    private static CallbackManager mCallbackManager = CallbackManager.Factory.create();

    protected static CallbackManager getCallbackManager() {
        return mCallbackManager;
    }

    private final ReactNativeHost host = new NavigationReactNativeHost(this) {
        @Override
        public boolean getUseDeveloperSupport() {
            return BuildConfig.DEBUG;
        }

        @Override
        protected String getJSBundleFile() {
            return CodePush.getJSBundleFile();
        }

        protected List<ReactPackage> getPackages() {
            // Add additional packages you require here
            // No need to add RnnPackage and MainReactPackage

            @SuppressWarnings("UnnecessaryLocalVariable")
            List<ReactPackage> packages = new PackageList(this).getPackages();
            // Packages that cannot be autolinked yet can be added manually here, for
            // example:

            packages.add(new LinearGradientPackage());
            packages.add(new KeyboardManagerPackage());
            return packages;
        }

        public List<ReactPackage> createAdditionalReactPackages() {
            return getPackages();
        }

        @Override
        protected String getJSMainModuleName() {
            return "index";
        }

    };

    @Override
    public void onCreate() {
        super.onCreate();

        /* Disable app being forced into Right-To-Left format */
        I18nUtil sharedI18nUtilInstance = I18nUtil.getInstance();
        sharedI18nUtilInstance.allowRTL(getApplicationContext(), false);

        initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
    }

    public boolean isDebug() {
        return BuildConfig.DEBUG;
    }

    @Override
    protected void attachBaseContext(Context context) {
        super.attachBaseContext(context);
        MultiDex.install(this);
    }

    @Override
    public ReactNativeHost getReactNativeHost() {
        return this.host;
    }

    /**
     * Loads Flipper in React Native templates. Call this in the onCreate method
     * with something like initializeFlipper(this,
     * getReactNativeHost().getReactInstanceManager());
     *
     * @param context
     * @param reactInstanceManager
     */
    private static void initializeFlipper(Context context, ReactInstanceManager reactInstanceManager) {
        if (BuildConfig.DEBUG) {
            try {
                /*
                 * We use reflection here to pick up the class that initializes Flipper, since
                 * Flipper library is not available in release mode
                 */
                Class<?> aClass = Class.forName("com.xxxxxxxx.ReactNativeFlipper");
                aClass.getMethod("initializeFlipper", Context.class, ReactInstanceManager.class).invoke(null, context,
                        reactInstanceManager);
            } catch (ClassNotFoundException e) {
                e.printStackTrace();
            } catch (NoSuchMethodException e) {
                e.printStackTrace();
            } catch (IllegalAccessException e) {
                e.printStackTrace();
            } catch (InvocationTargetException e) {
                e.printStackTrace();
            }
        }
    }
}

```

#### `AndroidManifest.xml`:

```xml
<?xml version="1.0" encoding="utf-8" ?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    package="com.xxxxxxxxxxx">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
    
    <!-- required for react-native-share base64 sharing -->
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.DOWNLOAD_WITHOUT_NOTIFICATION" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />

    <uses-feature android:name="android.hardware.camera" android:required="false" />
    <uses-feature android:name="android.hardware.camera.front" android:required="false" />

    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
    <!--  This permission is needed for Singular to retrieve Google Play Referrer data  -->
    <uses-permission android:name="com.google.android.finsky.permission.BIND_GET_INSTALL_REFERRER_SERVICE" />
    <!--  This permission is needed for Singular to retrieve data from the google licensing API  -->
    <uses-permission android:name="com.android.vending.CHECK_LICENSE" />

    <uses-permission android:name="android.permission.READ_CONTACTS" />

    <!--permissions to remove-->
    <uses-permission android:name="android.permission.READ_PHONE_STATE" android:maxSdkVersion="22"/>
    <uses-permission android:name="android.permission.USE_FINGERPRINT" tools:node="remove" />
    <uses-permission android:name="android.permission.USE_BIOMETRIC" tools:node="remove" />



    <application
        android:name=".MainApplication"
        android:allowBackup="false"
        android:icon="@mipmap/ic_launcher"
        android:label="@string/app_name"
        android:largeHeap="true"
        android:theme="@style/AppTheme"
        android:requestLegacyExternalStorage="true"
        tools:targetApi="q">
        <activity
            android:name=".MainActivity"
            android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
            android:label="@string/app_name"
            android:launchMode="singleTask"
            android:windowSoftInputMode="adjustUnspecified">
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
                <action android:name="android.intent.action.DOWNLOAD_COMPLETE" />
            </intent-filter>
            <intent-filter>
                <action android:name="android.intent.action.VIEW" />
                <category android:name="android.intent.category.DEFAULT" />
                <category android:name="android.intent.category.BROWSABLE" />
            </intent-filter>
            <!-- DON'T DELETE, NEEDED FOR SHARE EXTENSION -->
            <intent-filter>	
                <data android:scheme="xxxxxxxxx" />	
                <action android:name="android.intent.action.VIEW" />	
                <category android:name="android.intent.category.DEFAULT" />	
                <category android:name="android.intent.category.BROWSABLE" />	
            </intent-filter>

            <intent-filter>
                <action android:name="android.intent.action.SEND" />
                <category android:name="android.intent.category.DEFAULT" />
                <data android:mimeType="text/plain" />
            </intent-filter>
        </activity>
        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" /> <!-- Notification icon and color -->
        <meta-data
            android:name="com.google.firebase.messaging.default_notification_icon"
            android:resource="@drawable/ic_notification" />
        <meta-data
            android:name="com.google.firebase.messaging.default_notification_color"
            android:resource="@color/colorAccent" 
            tools:replace="android:resource"/>
        <!-- <meta-data
            android:name="com.google.firebase.messaging.default_notification_channel_id"
            android:value="@string/default_notification_channel_id"/> -->


        <meta-data
            android:name="com.facebook.sdk.ApplicationId"
            android:value="@string/facebook_app_id" />
        <meta-data
            android:name="com.facebook.sdk.CodelessDebugLogEnabled"
            android:value="true" />
        

        <!-- <service android:name="io.invertase.firebase.messaging.RNFirebaseMessagingService">
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service> -->


        <service android:name="io.invertase.firebase.messaging.RNFirebaseBackgroundMessagingService" />
    </application>

</manifest>
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

<!-- Please run `react-native info` on your terminal and paste the contents into the code block below -->

```
System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Memory: 5.13 GB / 32.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 15.6.0 - /usr/local/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 7.4.0 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.0 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.1, DriverKit 19.0, macOS 10.15, tvOS 14.0, watchOS 7.0
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 28.0.3, 29.0.2, 30.0.2
      System Images: android-29 | Android TV Intel x86 Atom, android-29 | Intel x86 Atom, android-29 | Intel x86 Atom_64, android-29 | Google APIs Intel x86 Atom, android-29 | Google APIs Intel x86 Atom_64, android-29 | Google Play Intel x86 Atom, android-29 | Google Play Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom, android-30 | Google Play Intel x86 Atom, android-30 | Google Play Intel x86 Atom_64
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.1 AI-201.8743.12.41.6858069
    Xcode: 12.1/12A7403 - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_282 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1
    react-native: 0.63.2 => 0.63.2
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

<!-- change `[ ]` to `[x]` to select an option(s) -->

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [x] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `10.5.0`
- **`Firebase` module(s) you're using that has the issue:**
  - `Crashlytics, DynamicLinks, InAppMessaging, Messaging, RemoteConfig`
- **Are you using `TypeScript`?**
  - `Y`

</p>
</details>

<!-- Thanks for reading this far down ❤️  -->
<!-- High quality, detailed issues are much easier to triage for maintainers -->

<!-- For bonus points, if you put a 🔥 (:fire:) emojii at the start of the issue title we'll know -->
<!-- that you took the time to fill this out correctly, or, at least read this far -->

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

--- a/docs/dynamic-links/usage/index.md
+++ b/docs/dynamic-links/usage/index.md
@@ -11,6 +11,8 @@ previous: /database/presence-detection
 This module requires that the `@react-native-firebase/app` module is already setup and installed. To install the "app" module, view the
 [Getting Started](/) documentation.
 
+This module also requires that the `@react-native-firebase/analytics` module is already setup and installed. To install the "analytics" module, view it's [Getting Started](/analytics/usage) documentation.
+
 ```bash
 # Install & setup the app module
 yarn add @react-native-firebase/app

--- a/docs/in-app-messaging/usage/index.md
+++ b/docs/in-app-messaging/usage/index.md
@@ -11,6 +11,8 @@ previous: /iid/usage
 This module requires that the `@react-native-firebase/app` module is already setup and installed. To install the "app" module, view the
 [Getting Started](/) documentation.
 
+This module also requires that the `@react-native-firebase/analytics` module is already setup and installed. To install the "analytics" module, view it's [Getting Started](/analytics/usage) documentation.
+
 ```bash
 # Install & setup the app module
 yarn add @react-native-firebase/app

--- a/docs/remote-config/usage/index.md
+++ b/docs/remote-config/usage/index.md
@@ -11,6 +11,8 @@ previous: /ml/image-labeling
 This module requires that the `@react-native-firebase/app` module is already setup and installed. To install the "app" module, view the
 [Getting Started](/) documentation.
 
+This module also requires that the `@react-native-firebase/analytics` module is already setup and installed. To install the "analytics" module, view it's [Getting Started](/analytics/usage) documentation.
+
 ```bash
 # Install & setup the app module
 yarn add @react-native-firebase/app

--- a/packages/dynamic-links/package.json
+++ b/packages/dynamic-links/package.json
@@ -23,7 +23,8 @@
     "dynamic link"
   ],
   "peerDependencies": {
-    "@react-native-firebase/app": "10.5.0"
+    "@react-native-firebase/app": "10.5.0",
+    "@react-native-firebase/analytics": "10.5.1"
   },
   "publishConfig": {
     "access": "public"

--- a/packages/in-app-messaging/package.json
+++ b/packages/in-app-messaging/package.json
@@ -27,7 +27,8 @@
     "inAppMessaging"
   ],
   "peerDependencies": {
-    "@react-native-firebase/app": "10.5.0"
+    "@react-native-firebase/app": "10.5.0",
+    "@react-native-firebase/analytics": "10.5.1"
   },
   "publishConfig": {
     "access": "public"

--- a/packages/remote-config/package.json
+++ b/packages/remote-config/package.json
@@ -24,7 +24,8 @@
     "remote-config"
   ],
   "peerDependencies": {
-    "@react-native-firebase/app": "10.5.0"
+    "@react-native-firebase/app": "10.5.0",
+    "@react-native-firebase/analytics": "10.5.1"
   },
   "publishConfig": {
     "access": "public"

